src/ppl/src/SimulatedAnnealing.cpp

```diff
@@ -121,6 +123,29 @@ void SimulatedAnnealing::run(float init_temperature,
       }
 
       temperature *= alpha_;
+
+      if (debug_->isOn()) {
+        bool isNoPauseMode = false;
+
+        if (debug_->isNoPauseMode_) {
+          isNoPauseMode = true;
+        }
+
+        if (iter == 0 || (iter + 1) % debug_->painting_interval_ == 0) {
```

Comment:
Too much logic here for what isn't the annealer's concern.  Just call the renderer and let it decide whether to draw or not.